### PR TITLE
feat(sentry-worker): sentry worker added to handle sentry-sdk events

### PR DESCRIPTION
**Pull request by n0str**
We added new worker to convert Sentry Envelope to Hawk Event.

## New worker — Sentry
- `sentry/src/index.ts` — decode envelope, parse it, convert all items to hawk event, add task to Grouper
- `sentry/src/utils/converter.ts` — converts Sentry event parts to Hawk event parts
- `sentry/playground` — sandbox to send errors via sentry nodejs sdk

### Logging Improvements:
* [`lib/logger.ts`](diffhunk://#diff-635cef180d112a3e6ac4deed3b9f46a4accedddbcb402ac1f9e25aca31de4b75R3-R74): Extended the `Logger` interface with a custom `json` method for logging JSON objects.
* [`lib/worker.ts`](diffhunk://#diff-1fb55b66996c5b21d6b82484086ad9e5a373b17f351b9fdd9f860109f059cbaeR194-R203): Added a `muteLogger` method to mute or unmute the logger, useful for tests expecting errors.

### Testing Configuration Updates:
* [`.eslintrc.js`](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5R8-R12): Added `package.json`, `tsconfig.json`, and `dist` to `ignorePatterns`.
* [`.github/workflows/tests.yml`](diffhunk://#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fL8-R12): Updated the runner to `ubuntu-20.04` and `setup-node` action to `v4`.
* [`jest-mongodb-config.js`](diffhunk://#diff-6873ad99f04de5db1c9e21750b711593244e44747233fd61c68f3efc9c74b77aL8-R8): Updated MongoDB version to `6.0.2`.
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R22): Added new test and run scripts for Sentry worker and updated `@shelf/jest-mongodb` to `^4.3.2`. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R22) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R32) [[3]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L70-R72)

### CODE REVIEW

File: workers/sentry/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+   * @param event - event to handle
+   */
+  public async handle(event: SentryEventWorkerTask): Promise<void> {
+    /**
+     * Define  event type
+     */
+    this.type = 'errors/sentry';
+
+    const rawEvent = b64decode(event.payload.envelope);
+    const envelope = this.parseSentryEnvelope(rawEvent);
```
Comment (neSpecc) [inline workers/sentry/src/index.ts:31]: add try-catch, please

File: workers/sentry/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    this.logger.debug(JSON.stringify(envelope));
+
+    // Todo: For now, we only handle the first item in the envelope
+    const hawkEvent = this.transformToHawkFormat(envelope.Header, envelope.Items[0], event.projectId);
```
Comment (neSpecc) [inline workers/sentry/src/index.ts:35]: why don't just call this in a loop, do handle all items?

File: workers/sentry/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    delete trace.public_key;
+
+    // convert sent_at from ISO 8601 to Unix timestamp
+    const sent_at_unix = Math.floor(new Date(sent_at).getTime() / 1000);
+
+    const backtrace = Payload.exception?.values?.[0]?.stacktrace?.frames?.map((frame: any) => ({
+      file: frame.filename,
+      line: frame.lineno,
+      function: frame.function,
+      sourceCode: frame.pre_context?.concat(frame.context_line, frame.post_context || [])
```
Comment (neSpecc) [inline workers/sentry/src/index.ts:107]: why there are only 5 rows above? where is the 5 rows below? I think, sentry should send them as well

image

File: workers/sentry/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        title: `${Payload.exception?.values?.[0]?.type || 'Unknown'}: ${Payload.exception?.values?.[0]?.value || ''}`,
+        type: Payload.level || 'error',
+        timestamp: sent_at_unix,
+        backtrace,
+        release: trace?.release || null,
+        context: {
+          sentAt: sent_at_unix,
+          trace: trace || {},
+          environment: trace?.environment || 'unknown',
+        },
```
Comment (neSpecc) [inline workers/sentry/src/index.ts:127]: do sentry sends additional data like our "context" field?

File: workers/sentry/src/index.ts (hunk 3 above)
Comment (n0str) [reply workers/sentry/src/index.ts:130]: It's okay. This just the last line of the script :) if you run script with error on not the last row – there will be 5 rows after
image

File: workers/sentry/src/index.ts (hunk 4 above)
Comment (n0str) [reply workers/sentry/src/index.ts:150]: yep! Context is sent via trace param.
```
context: {
          sentAt: sent_at_unix,
          trace: trace || {},
          environment: trace?.environment || 'unknown',
        },
```

trace sample is on the screenshot
image

**Review by neSpecc — APPROVED** [review]

**Pull request merged**